version_test.go

```diff
@@ -31,3 +25,55 @@ func TestDockerfileVersion(t *testing.T) {
 		t.Errorf("didn't find %q in Dockerfile", wantSub)
 	}
 }
+
+// TestGoVersion tests that the Go version specified in go.mod matches ./tool/go version.
+func TestGoVersion(t *testing.T) {
+	goModVersion := mustGetGoModVersion(t, true)
+
+	goToolCmd := exec.Command("./tool/go", "version")
+	goToolOutput, err := goToolCmd.Output()
+	if err != nil {
+		t.Fatalf("Failed to get ./tool/go version: %v", err)
+	}
+
+	// Version info would approximately like 'go version go1.24.4 linux/amd64'.
+	parts := strings.Fields(string(goToolOutput))
+	if len(parts) < 4 {
+		t.Fatalf("Unexpected ./tool/go version output format: %s", goToolOutput)
+	}
+
+	goToolVersion := strings.TrimPrefix(parts[2], "go")
+
+	if goModVersion != goToolVersion {
+		t.Errorf("Go version in go.mod (%s) does not match the version of ./tool/go (%s).\nEnsure that the go.mod refers to the same Go version as ./go.toolchain.rev.",
```

Comment:
%q for any strings instead of %s .... %q shows trailing whitespace, binary, etc.